src/core/memory.d

```diff
@@ -881,36 +873,69 @@ void pureFree(void* ptr) @system pure @nogc nothrow
 
 @system pure nothrow @nogc unittest
 {
-    const int errno = fakePureGetErrno();
+    const int errno = fakePureErrno();
 
     void* x = pureMalloc(10);            // normal allocation
-    assert(errno == fakePureGetErrno()); // errno shouldn't change
+    assert(errno == fakePureErrno()); // errno shouldn't change
     assert(x !is null);                   // allocation should succeed
 
     x = pureRealloc(x, 10);              // normal reallocation
-    assert(errno == fakePureGetErrno()); // errno shouldn't change
+    assert(errno == fakePureErrno()); // errno shouldn't change
     assert(x !is null);                   // allocation should succeed
 
     fakePureFree(x);
 
     void* y = pureCalloc(10, 1);         // normal zeroed allocation
-    assert(errno == fakePureGetErrno()); // errno shouldn't change
+    assert(errno == fakePureErrno()); // errno shouldn't change
     assert(y !is null);                   // allocation should succeed
 
     fakePureFree(y);
 
     // subtract 2 because snn.lib adds 2 unconditionally before passing
     //  the size to the Windows API
     void* z = pureMalloc(size_t.max - 2); // won't affect `errno`
-    assert(errno == fakePureGetErrno()); // errno shouldn't change
+    assert(errno == fakePureErrno()); // errno shouldn't change
     assert(z is null);
 }
 
 // locally purified for internal use here only
 extern (C) private pure @system @nogc nothrow
 {
-    pragma(mangle, "getErrno") int fakePureGetErrno();
-    pragma(mangle, "setErrno") int fakePureSetErrno(int);
+    /* see core.stdc.errno for errno function declarations
+     */
+
+    version (CRuntime_DigitalMars)
+    {
+        pragma(mangle, "_errno") ref int fakePureErrno();
+    }
+    else version (CRuntime_Microsoft)
+    {
+        pragma(mangle, "_errno") ref int fakePureErrno();
+    }
+    else version (CRuntime_Glibc)
+    {
+        pragma(mangle, "__errno_location") ref int fakePureErrno();
+    }
+    else version (FreeBSD)
+    {
+        pragma(mangle, "__error") ref int fakePureErrno();
+    }
+    else version (linux)
+    {
+        pragma(mangle, "__errno_location") ref int fakePureErrno();
+    }
+    else version (Darwin)
+    {
+        pragma(mangle, "__error") ref int fakePureErrno();
+    }
+    else version (OSX)
+    {
+        pragma(mangle, "__error") ref int fakePureErrno();
+    }
+    else
+    {
+        static assert(0, "add system errno function");
+    }
```

Comment:
You can avoid repeating all the platform details by wrapping it in a function (similar as pureReprintReal in core.demangle):
```
extern(C) private {
    pure @trusted @nogc nothrow pragma(mangle, "fakePureErrno") ref int pureErrno();
    ref int fakePureErrno() { return errno(); }
}
```

Reply:
Good idea.